File: Loom.Core/Models/LlmResponse.vb
Namespace Models

    ''' <summary>
    ''' Properties of LLM response
    ''' </summary>
    Public Class LlmResponse
        Public Property Id As String
        Public Property ModelUsed As String
        Public Property Role As Enums.MessageRole
        Public Property Content As String
        Public Property ToolCalls As List(Of ToolCallInvocation)



        Public Property PromptTokens As Integer
        Public Property CompletionTokens As Integer
        Public Property TotalTokens As Integer


        Public Property Metadata As New Dictionary(Of String, Object)
    End Class
End Namespace

File: implementations/csharp/src/Loom.Core/Models/LlmResponse.cs
using System.Collections.Generic;

namespace Loom.Core.Models
{
    /// <summary>
    /// Properties of LLM response
    /// </summary>
    public class LlmResponse
    {
        public string Id { get; set; }
        public string ModelUsed { get; set; }
        public string Content { get; set; }
        public List<ToolCallInvocation> ToolCalls { get; set; }
        public Enums.MessageRole Role { get; set; }

        public int PromptTokens { get; set; }
        public int CompletionTokens { get; set; }
        public int TotalTokens { get; set; }

        public Dictionary<string, object> Metadata { get; set; }
    }
}
